src/decays/decay.cpp

```diff
@@ -133,4 +155,174 @@ double hVV_4body(double *q2, size_t /* dim */, void *params)
+         effC.particle = p;
+         if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::gg])) {
+            effC.gg = c;
+         }
+         else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::gamgam])) {
+            effC.gamgam = c;
+         }
+         else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::ZZ])) {
+            effC.ZZ = c;
+         }
+         else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::WW])) {
+            effC.WW = c;
+         }
+         else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::Zgam])) {
+            effC.Zgam = c;
+         }
+         effective_coupling_list.push_back(std::move(effC));
+      }
+      else {
+         if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::gg])) {
+            found->gg = c;
+         }
+         else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::gamgam])) {
+            found->gamgam = c;
+         }
+         else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::ZZ])) {
+            found->ZZ = c;
+         }
+         else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::WW])) {
+            found->WW = c;
+         }
+         else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::Zgam])) {
+            found->Zgam = c;
+         }
+      }
+   }
+
+void EffectiveCoupling_list::add_coupling(std::string const& p, std::array<int, 2> const& fs, std::pair<std::string, std::complex<double>> const& c) {
+
+   auto found = std::find_if(
+      std::begin(effective_coupling_list), std::end(effective_coupling_list),
+      [&p](NeutralHiggsEffectiveCouplings const& effC) {return effC.particle == p;}
+   );
+
+   auto are_the_same = [](std::array<int, 2> const& a1, std::array<int, 2> const& a2) {
+         return std::is_permutation(a1.begin(), a1.end(), a2.begin(), a2.end());
+   };
+   auto negate = [](std::array<int, 2> in) {
+      std::for_each(in.begin(), in.end(), [](int& el){el *= -1; });
+      return in;
+   };
+
+   if (found == std::end(effective_coupling_list)) {
+      auto effC = NeutralHiggsEffectiveCouplings {};
+      effC.particle = p;
+      if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::dd])) {
+         effC.dd = c;
+      }
+      else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::uu])) {
+         effC.uu = c;
+      }
+      else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::ss])) {
+         effC.ss = c;
+      }
+      else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::cc])) {
+         effC.cc = c;
+      }
+      else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::bb])) {
+         effC.bb = c;
+      }
+      else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::tt])) {
+         effC.tt = c;
+      }
+      else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::ee])) {
+         effC.ee = c;
+      }
+      else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::mumu])) {
+         effC.mumu = c;
+      }
+      else if (are_the_same(fs, pdg_id_pairs[PDG_id_pairs::tautau])) {
+         std::cout << fs[0] << ' ' << fs[1] << std::endl;
```

Comment:
It seems that this is a left-over comment from debugging. I guess it should be removed.